### ISSUE TITLE
GC.ReRegisterForFinalize(Object) Method – object argument/param is not limited to caller

### ISSUE THREAD

**Issue by Uzivatel919** [dotnet/docs#26061]
From [Remarks](https://docs.microsoft.com/en-us/dotnet/api/system.gc.reregisterforfinalize?view=net-5.0#remarks)
> The ReRegisterForFinalize method adds the obj parameter to the list of objects that request finalization before the garbage collector frees the object. The obj parameter must be the caller of this method.

Verifying example one can gather no error on `GC.ReRegisterForFinalize (new object());` call.
```csharp
class Program
{
  static async Task Main ( string[] args )
  {
    Test test;
    foreach(var i in Enumerable.Range(0, 2))
      test = new Test ();
    
    test = null;
    GC.Collect ();
    
    await Task.Delay (1_000);
  }
  
  class Test
  {
    ~Test ()
    {        
      Console.WriteLine ("AAAA");        
      GC.ReRegisterForFinalize (this);
      GC.ReRegisterForFinalize (new object());
    }
  }
}
```

### Project
[GcReRegisterForFinalize.zip](https://github.com/dotnet/docs/files/7148204/GcReRegisterForFinalize.zip)

**Comment by svick** [comment]
I don't know if the documentation is correct here, but your code does not necessarily prove that it's incorrect. The quoted documentation does not say what happens if you call this method in a way that violates its contract, which means this is undefined behavior, and "working as expected" is consistent with that.

**Comment by Uzivatel919** [comment]
@svick, yes, yes, it seems possible that violation does not come from runtime… Let consult another quote stated at end of [Remarks](https://docs.microsoft.com/en-us/dotnet/api/system.gc.reregisterforfinalize?view=net-5.0#remarks)
> A finalizer can use this method to resurrect itself or an object that it references.

This statement confirms ability to _resurrect_ any object.

**Comment by BillWagner** [comment]
transferring to dotnet-api-docs
